render/helpers.go

```diff
@@ -145,10 +145,25 @@ func DatePeriodFormat(s string) string {
 	if _, err := strconv.Atoi(s[:4]); err == nil {
 		// Not just displaying year but month as well
 		if len(s) > 5 {
-			// YYYY[space] = 5 characters
-			postYearIndex := 5
-			s = s[postYearIndex:] + " " + s[:4]
-
+			if len(s) < 14 {
+				// YYYY[space] = 5 characters
+				postYearIndex := 5
+				s = s[postYearIndex:] + " " + s[:4]
+			}
+			if len(s) == 14 {
+				year, _ := strconv.Atoi(s[:4])
```

Comment:
`strconv.Atoi(s[:4])` is previously called in line 145 so I'd suggest extracting it to a local variable instead of making an extra conversion?